pkg/termination/termination_test.go

```diff
@@ -0,0 +1,364 @@
+/*
+Copyright The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package termination
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"sync/atomic"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	machinev1 "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/klogr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var emptyEvents = func(rw http.ResponseWriter, req *http.Request) {
+	rw.Write([]byte(`{"DocumentIncarnation":0,"Events":[]}`))
+}
+
+var _ = Describe("Handler Suite", func() {
+	var terminationServer *httptest.Server
+	var httpHandler http.Handler
+	var nodeName string
+	var stop chan struct{}
+	var errs chan error
+	var h *handler
+
+	BeforeEach(func() {
+		// Reset test vars
+		terminationServer = nil
+		httpHandler = nil
+		nodeName = "test-node"
+		httpHandler = newMockHTTPHandler(emptyEvents)
+		stop = nil
+		errs = nil
+
+		// use NewHandler() instead of manual construction in order to test NewHandler() logic
+		// like checking that machine api is added to scheme
+		handlerInterface, err := NewHandler(klogr.New(), cfg, 100*time.Millisecond, "", nodeName)
+		Expect(err).ToNot(HaveOccurred())
+
+		h = handlerInterface.(*handler)
+
+		// set pollURL so we can override initial value later
+		h.pollURL = nil
+	})
+
+	JustBeforeEach(func() {
+		Expect(httpHandler).ToNot(BeNil())
+		terminationServer = httptest.NewServer(httpHandler)
+
+		if h.pollURL == nil {
+			pollURL, err := url.Parse(terminationServer.URL)
+			Expect(err).ToNot(HaveOccurred())
+			h.pollURL = pollURL
+		}
+	})
+
+	AfterEach(func() {
+		if stop != nil && !isClosed(stop) {
+			close(stop)
+		}
+		terminationServer.Close()
+
+		Expect(deleteAllMachines(k8sClient)).To(Succeed())
+	})
+
+	Context("when running the handler", func() {
+		JustBeforeEach(func() {
+			stop, errs = StartTestHandler(h)
+		})
+
+		Context("when the handler is stopped", func() {
+			JustBeforeEach(func() {
+				close(stop)
+			})
+
+			It("should not return an error", func() {
+				Eventually(errs).Should(Receive(BeNil()))
+			})
+		})
+
+		Context("when no machine exists for the node", func() {
+			It("should not receive an error upon starting", func() {
+				Eventually(errs).ShouldNot(Receive())
```

Comment:
This should use a `Consistently` instead
```suggestion
				Consistently(errs).ShouldNot(Receive())
```